### PR TITLE
Add Sphinx page for API

### PR DESCRIPTION
**Pull request by bmwant**
## Rationale 
We need to have better documentation for functions that package provides. This PR addresses  #157 issue.

## Changes
* Add `api` page with some usage examples
* Update travis button to correctly redirect to Travis page

## Testing
image

@asvetlov is there anything like staging(?) for the documentation where other users can check how it looks for such PRs with documentation updates?

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+   :param str app_key: key that will be used to access templating environment
+                       from application dictionary object.
+
+   :param context_processors: list of :ref:`aiohttp-web-middlewares`. These are
+                              context processors used to rewrite or inject some
+                              variables during the processing of a request.
+   :type context_processors: :class:`list`
+
+   :param filters: desc.
```
Comment (asvetlov) [inline docs/api.rst]: Did you miss description?

Comment (bmwant) [reply docs/api.rst]: yes, it's still work in progress. I'm wondering whether it's something that was requested by the issue or you expected something else within the scope of this task?

Comment (asvetlov) [reply docs/api.rst]: `__all__ = ('setup', 'get_env', 'render_template', 'template')`.
`get_env` and `template` should be documented (as well as installed context functions),
`render_string` should be added to `__all__`.

**Comment by bmwant** [comment]
@asvetlov could you check it once again? :smile:

**Comment by asvetlov** [comment]
Will do, or please remind me in two weeks.
Sorry, I'm overwhelmed at the moment.

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+.. module:: aiohttp_jinja2
+.. highlight:: python
+
+
+.. function:: setup(app, *args, app_key=APP_KEY, context_processors=(),
+                    filters=None, default_helpers=True, **kwargs)
+
+   Function responsible for initializing templating system on application. This
+   one is required to be called first in order to start using this package with
+   your application.
```
Comment (samuelcolvin) [inline docs/api.rst]: improving the english slightly:

```
Function responsible for initializing templating system on an application. This
￼much be called before freezing or running the application in order to use *aiohttp-jinja*.
```

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+   :param app: :class:`aiohttp.web.Application` instance to initialize template
+               system on.
+
+   :param str app_key: optional key that will be used to access templating
+                       environment from application dictionary object.
```
Comment (samuelcolvin) [inline docs/api.rst]: maybe adds `defaults to ...`

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+   :param context_processors: list of :ref:`aiohttp-web-middlewares`. These are
+                              context processors used to rewrite or inject some
+                              variables during the processing of a request.
+   :type context_processors: :class:`list`
+
+   :param filters: list of functions allowing to modify variables within a
+                   template.
```
Comment (samuelcolvin) [inline docs/api.rst]: ```extra jinja filters [link to jinja filter documentation]```

File: docs/api.rst [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+   :param app: :class:`aiohttp.web.Application` instance to get variables from.
+
+   :param str app_key: optinal key that will be used to access templating
+                       environment from application dictionary object.
+
+
+.. function:: template(template_name, *, app_key, encoding, status)
+
+   Behaves as a decorator around view functions accepting template name that
+   should be rendered as a result. Supports both synchronous and asynchronous
```
Comment (samuelcolvin) [inline docs/api.rst]: "that should be used to render the response"

File: docs/api.rst [hunk 6]
```diff
@@ -0,0 +1,128 @@
+API
+===
+
+This page describes module API with detailed explanations of the functions
```
Comment (samuelcolvin) [inline docs/api.rst]: "Describes the module API with detailed explanations of functions parameters."

File: docs/api.rst [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+   :param str app_key: optional key that will be used to access templating
+                       environment from application dictionary object.
+
+   :param str encoding: encoding that will be set as a charset property on the
+                        response for rendered template, default to utf-8.
+
+   :params int status: http status code that will be set on resulting response.
+
+
+Example of usage
```
Comment (samuelcolvin) [inline docs/api.rst]: do we need this given that similar examples area already given else where in the docs?

File: docs/api.rst [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+.. function:: render_string(template_name, request, context, *,
+                            app_key=APP_KEY)
+
+   Renders template specified and returns resulting string.
+
+   :param str template_name: Name of the template you want to render. Usually
+                             it's a filename without extension on your
+                             filesystem.
+   :param request: request to the main application that implies template
```
Comment (samuelcolvin) [inline docs/api.rst]: not sure what you mean by "implies", maybe "aiohttp request associated with an application where aiohttp-jinja rendering is configured"?

File: docs/api.rst [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+                   rendering.
+   :type request: :class:`aiohttp.web.Request`
+
+   :param context: set of variables that are used to fill the template.
```
Comment (samuelcolvin) [inline docs/api.rst]: using "set" is confused as it's not a `set`. Maybe "dict used as context when rendering the template"

File: docs/api.rst [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+   :param str app_key: is an optional key for application dict.
+
+
+.. function:: render_template(template_name, request, context, *,
+                              app_key=APP_KEY, encoding='utf-8', status=200)
+
+   :param str template_name: Name of the template you want to render.
+   :param request: request to the main application that implies template
```
Comment (samuelcolvin) [inline docs/api.rst]: as above.

File: docs/api.rst [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+                   rendering.
+   :type request: :class:`aiohttp.web.Request`
+
+   :param dict context: set of variables that are used to fill the template.
```
Comment (samuelcolvin) [inline docs/api.rst]: as above.

File: docs/api.rst [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+   :param app_key: is an optional key for application dict.
+   :param int status: desc.
```
Comment (samuelcolvin) [inline docs/api.rst]: this needs filling in.

**Comment by bmwant** [comment]
@samuelcolvin wow, thanks for such a thorough review. I hope this time it's much better, waiting for your review once again.

**Comment by asvetlov** [comment]
@samuelcolvin is something left to do?

**Comment by asvetlov** [comment]
Merging, docs in PR is better than current anyway

**Pull request merged**